scripts/manage_demux_stats.py

```diff
@@ -329,7 +329,26 @@ def set_sample_values(demux_process, parser_struct, process_stats):
         # Artifacts in each lane
         for target_file in outarts_per_lane:
             try:
-                current_name = target_file.samples[0].name
+                # This block addresses a LIMS bug in which multiple samples are tied to the same demux artifact
+                # In this case, try to find a single sample name matching the name of the demux artifact
+                if len(target_file.samples) > 1:
+                    matching_names = [
+                        sample.name
+                        for sample in target_file.samples
+                        if sample.name in target_file.name
+                    ]
```

Comment:
So this is to make sure that the scripts selects the same single sample as the fix_demux_artifact_names script? Because even without this fix, there will be a single sample name selected right?

Reply:
Yes, the current code assumes each demux artifact is tied to a single sample, which is not true for this bug, causing the script to populate the LIMS fields rather haphazardly.

The new code adds a logic block to address this case, and tries to rely on the demux artifact name having been overwritten by `fix_demux_artifact_names` to point to the single correct sample.

We could have everything addressed in this script instead of having two separate EPPs, if we want, but that would probably involve some major refactoring and warrant a validation.